host_source.go

```diff
@@ -388,6 +392,10 @@ func (h *HostInfo) HostnameAndPort() string {
 	return net.JoinHostPort(h.hostname, strconv.Itoa(h.port))
 }
 
+func (h *HostInfo) Endpoint() *Endpoint {
+	return h.endpoint
+}
```

Comment:
Typically those methods have mutex protected access to variables.

Reply:
Due to update?

Reply:
it's not updated anywhere, but i added rlock.